e2e/portalicious/tests/DoPayment/DoSuccessfulPayment.spec.ts

```diff
@@ -0,0 +1,71 @@
+import { test } from '@playwright/test';
+import { format } from 'date-fns';
+
+import { SeedScript } from '@121-service/src/scripts/seed-script.enum';
+import NLRCProgram from '@121-service/src/seed-data/program/program-nlrc-ocw.json';
+import { seedIncludedRegistrations } from '@121-service/test/helpers/registration.helper';
+import { resetDB } from '@121-service/test/helpers/utility.helper';
+import { getAccessToken } from '@121-service/test/helpers/utility.helper';
+import { registrationsOCW } from '@121-service/test/registrations/pagination/pagination-data';
+
+import BasePage from '@121-e2e/portalicious/pages/BasePage';
+import LoginPage from '@121-e2e/portalicious/pages/LoginPage';
+import PaymentsPage from '@121-e2e/portalicious/pages/PaymentsPage';
+
+test.beforeEach(async ({ page }) => {
+  await resetDB(SeedScript.nlrcMultiple);
+  const programIdOCW = 3;
+  const OcwProgramId = programIdOCW;
+
+  const accessToken = await getAccessToken();
+  await seedIncludedRegistrations(registrationsOCW, OcwProgramId, accessToken);
+
+  // Login
+  const loginPage = new LoginPage(page);
+  await page.goto('/');
+  await loginPage.login(
+    process.env.USERCONFIG_121_SERVICE_EMAIL_ADMIN,
+    process.env.USERCONFIG_121_SERVICE_PASSWORD_ADMIN,
+  );
+});
+
+test('[31970] Do successful payment', async ({ page }) => {
+  const basePage = new BasePage(page);
+  const payments = new PaymentsPage(page);
+
+  const projectTitle = 'NLRC OCW Program';
+  const numberOfPas = registrationsOCW.length;
+  const defaultTransferValue = NLRCProgram.fixedTransferValue;
+  const defaultMaxTransferValue = registrationsOCW.reduce((output, pa) => {
+    return output + pa.paymentAmountMultiplier * defaultTransferValue;
+  }, 0);
+  const date = new Date();
+  const formattedDate = format(date, 'dd/MM/yyyy, HH:mm');
+  const lastPaymentDate = `${formattedDate}`;
+
+  await test.step('Navigate to Program payments', async () => {
+    await basePage.selectProgram(projectTitle);
+
+    await payments.navigateToProgramPage('Payments');
+  });
+
+  await test.step('Do payment', async () => {
+    await payments.createPayment();
+    await payments.startPayment();
+  });
+
+  await test.step('Validate payment card', async () => {
+    await payments.validateToastMessage('Payment created.');
+    await payments.navigateToProgramPage('Payments');
+    // Reload the page in case there are still some payments in progress
+    await page.reload();
```

Comment:
In the frontend we have built logic to make sure that the "in progress" marker goes away after some time, once the payment is not in progress anymore. I think it would be better to wait for that in the test too, instead of reloading the page.

Reply:
If there is a "payment in progress":
- we should show a chip on this card
- we should keep hitting this endpoint to update the stats

So that a page reload is not needed.

(By the way, a page reload should never be needed, so this is good to know in general. If you are doing a page reload in a test, it might be a bug in the frontend.)

If this doesn't work as expected hit me up and we can have a look together.

Reply:
In Portal reload had to be used since there were no spinners. I see that we want to avoid it in Portalicious. I think there is a bug then. Because as in attached screenshot the spinner does not appear. I believe it is because of the use of multiple FSPs where one PA is has voucher instead of "Visa".

Reply:
I have fixed it according to the requirements. Now Playwright should wait for the chip with the spinner when payment is in progress.